chainer/functions/evaluation/binary_accuracy.py

```diff
@@ -27,7 +29,7 @@ def forward(self, inputs):
         t = t.ravel()
         c = (y >= 0)
         count = xp.maximum(1, (t != self.ignore_label).sum())
-        return xp.asarray((c == t).sum(dtype='f') / count, dtype='f'),
+        return xp.asarray((c == t).sum() / count, dtype=y.dtype),
```

Comment:
I think it is a division of an integer by an integer. So it causes a rounding to the integer, doesn't it?